### PR TITLE
feat(action): handle user cancellation

### PR DESCRIPTION
**Pull request by soonum**
This is useful in start mode where one can cancel before the instance is actually ready. This way a best-effort is produced to clean up any spawned instance instead of it turning into a zombie.

### CODE REVIEW

File: src/slab.js [hunk 1]
```diff
@@ -127,6 +127,9 @@ async function waitForInstance(taskId, taskName) {
       const task_status = body[taskName].status.toLowerCase()
 
       if (task_status === 'done') {
+        if (taskName === 'start') {
+          await acknowledgeTaskDone(taskId)
+        }
```
Comment (IceTDrinker) [inline src/slab.js:132]: did we store the runner name at this point ?

Comment (soonum) [reply src/slab.js:132]: Yes. It's done right after start request.

Comment (IceTDrinker) [inline src/slab.js:132]: so the process is:

ask for start -> slab starts instance, starts watchdog -> action waits for instance -> acknowledge ?

isn't there a risk of slab killing the instance before the action can aknowledge ?

Comment (IceTDrinker) [reply src/slab.js:132]: also is there a risk of aknowledging before Slab is waiting for an aknwoledge ?

Comment (soonum) [reply src/slab.js:132]: > isn't there a risk of slab killing the instance before the action can aknowledge ?

Reasonably, no. The sleeping time before killing the instance is 5 minutes on Slab side, AFTER the spawn is over.


> also is there a risk of aknowledging before Slab is waiting for an aknwoledge ?

No this is enforced on Slab side, you cannot acknowledge an InProgress or a Failed task (it's basically a no-op)

Comment (IceTDrinker) [reply src/slab.js:132]: is it possible for the action to acknowledge an in progress task ?

and if so, can it say "ok I acknowledged, now I do something else" and never acknowledges again, meaning slab could kill the instance ?

Comment (soonum) [reply src/slab.js:132]: Nope it's not possible to acknowledge an in progress task. The implementation imposes the acknowledge call to happen after a Done task has been received.

**Review by IceTDrinker — APPROVED** [review]

**Pull request merged**